src/test/java/com/venafi/vcert/sdk/connectors/tpp/TppTokenConnectorAT.java

```diff
@@ -39,40 +42,71 @@
     @BeforeEach
```

Comment:
I think if switch @BeforeEach with @BeforeAll it will not be needed to check whether is already called

Reply:
the @BeforeAll annotation cannot be used because there are specific tests that set invalid information in the credentials (like revoke test or invalid test). And we need that invalid data to be reset for the next test. However the purpose of the token is for all operations to be executed with the same token itself. Thats the reason we validate that if the info is set, do not request a new token. While negative use cases set that info to null once finished.